bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/internal/thing/ThingResource.java

```diff
@@ -350,6 +362,15 @@ public Response update(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(value
         // check configuration
         thingBean.configuration = normalizeConfiguration(thingBean.configuration, thing.getThingTypeUID(),
                 thing.getUID());
+        if (thingBean.channels != null) {
```

Comment:
This code is very similar to the new block at lines 186 - 193. This could be refactored into a helper method.